client/cz-component-dash.html

```diff
@@ -104,57 +104,74 @@
           type: 'Object',
           value: function() { return []; },
         },
-        component: {
+        updateSLO: {
           type: 'Object',
-          value: function() { return {}; },
+          value: function() { return null; },
         },
-        updateSLO: {
+        issueList: {
           type: 'Object',
-          value: function() { return {}; },
+          value: function() { return null; },
+        },
+        issueBreakdown: {
+          type: 'Object',
+          value: function() { return {}; }
         }
       },
 
+      observers: [
+        'updateIssueBreakdown(issueList)',
+      ],
+
       attached: function() {
-        registerSource('cz-config', 'users', function(users) {
+        registerSource('cz-config', 'users', users => {
           this.set('users', users.map(userData => userData['email']));
-        }.bind(this));
-        registerSource('cz-config', 'components', function(components) {
-          this.set('component', {name: components[this.key]});
-          this.updateComponentIssues(this.component)
-        }.bind(this));
-        registerSource('cz-config', 'updateSLO', function(updateSLO) {
+        });
+        registerSource('cz-config', 'updateSLO', updateSLO => {
           this.set('updateSLO', updateSLO);
-        }.bind(this));
+        });
+        registerSource('cz-config', 'issue-sources', issueSources => {
+          this.updateIssueSource(issueSources[this.key]);
+        });
       },
 
       summary: function(issueList, updateSLO) {
+        if (!issueList || !updateSLO) {
+          return [];
+        }
         return issueList.summary(updateSLO);
       },
 
-      updateComponentIssues: function(component) {
-        registerSource('cz-crbug-issues', {component: component.name}, function(issueList) {
-          var teamsIssueList = new IssueList();
-          var othersIssueList = new IssueList();
-          var unownedIssueList = new IssueList();
+      updateIssueSource: function({tag, query}) {
```

Comment:
(nit) splitting this out does not improve readability, but does destroy locality of effect